modules/social_features/social_album/social_album.module

```diff
@@ -0,0 +1,351 @@
+}
+
+/**
+ * Implements hook_entity_view().
+ */
+function social_album_entity_view(array &$build, EntityInterface $entity, EntityViewDisplayInterface $display, $view_mode) {
+  if ($display->getComponent('social_album_images')) {
+    $build['social_album_images'] = views_embed_view('albums', 'embed_album_overview');
+  }
+}
+
+/**
+ * Implements hook_preprocess_HOOK().
+ */
+function social_album_preprocess_block__social_post(&$variables) {
+  if (
+    \Drupal::routeMatch()->getRouteName() === 'social_album.post' &&
+    ($content = \Drupal::service('social_post.helper')->buildCurrentUserImage())
+  ) {
+    $variables['content'] = [
+      'form' => $variables['content'],
+      'current_user_image' => $content,
+    ];
+  }
+}
+
+/**
+ * Implements hook_preprocess_HOOK().
+ */
+function social_album_preprocess_node(&$variables) {
+  /** @var \Drupal\node\NodeInterface $node */
+  $node = $variables['node'];
+
+  if (
+    $node->bundle() === 'album' &&
+    $variables['view_mode'] === 'teaser' &&
+    views_get_view_result('albums', 'embed_album_cover', $node->id())
+  ) {
+    $variables['node_image'] = views_embed_view('albums', 'embed_album_cover', $node->id());
+    $variables['no_image'] = FALSE;
+  }
+}
+
+/**
+ * Implements hook_preprocess_HOOK().
+ */
+function social_album_preprocess_views_view(&$variables) {
+  /** @var \Drupal\views\ViewExecutable $view */
+  $view = $variables['view'];
+
+  if ($view->id() === 'albums') {
+    $library = [
+      'embed_album_overview' => 'social_album/social_album_post',
+      'page_albums_overview' => 'social_album/social_album_teaser',
+    ];
+
+    if (isset($library[$view->current_display])) {
+      $variables['#attached']['library'][] = $library[$view->current_display];
+    }
+  }
+}
+
+/**
+ * Implements hook_preprocess_HOOK().
+ */
+function social_album_preprocess_social_album_post(&$variables) {
+  $items = [];
+
+  foreach ($variables['items'] as $item) {
+    if (!isset($items[$pid = $item['pid']])) {
+      $parent_variables = ['pid' => $pid];
+      social_post_album_preprocess_album_post_popup($parent_variables);
+      $items[$pid] = ['post' => $parent_variables['post']];
+    }
+
+    $items[$pid]['urls'][] = $item['url'];
+  }
+
+  $variables['items'] = $items;
+}
+
+/**
+ * Implements hook_ENTITY_TYPE_insert().
+ */
+function social_album_post_insert(EntityInterface $entity) {
+  /** @var \Drupal\social_post\Entity\PostInterface $entity */
+  if ($entity->bundle() === 'photo' && !$entity->field_album->isEmpty()) {
+    \Drupal::service('cache_tags.invalidator')->invalidateTags([
+      'node:' . $entity->field_album->entity->id(),
+      'user:' . $entity->getOwnerId(),
+    ]);
+  }
+}
+
+/**
+ * Implements hook_ENTITY_TYPE_create_access().
+ */
+function social_album_node_create_access(AccountInterface $account, array $context, $entity_bundle) {
+  if (
+    $entity_bundle === 'album' &&
+    \Drupal::routeMatch()->getRouteName() === 'view.albums.page_albums_overview' &&
```

Comment:
In this case, I think we can use a variable (`$route_match`) for `\Drupal::routeMatch()` because it calls twice.